plugins/modules/cp4s/cp4s_query_incidents.py

```diff
@@ -0,0 +1,262 @@
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+from __future__ import (absolute_import, division, print_function)
+from ansible.module_utils.basic import AnsibleModule
+
+
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: cp4s_query_incidents
+
+short_description: A Module used to query Cases/Incidents in CP4S or Resilient
+
+# If this is part of a collection, you need to use semantic versioning,
+# i.e. the version is of the form "2.5.0" and not "2.4".
+version_added: "1.0.0"
+
+description: This module is an example of how you can choose to use a module or a role to achieve a similar outcome. An almost identical piece of functionality exists in the CP4S role but this gives a programmatic way to do it.
+
+options:
+    conditions:
+        description:
+            - list of conditions to query
+            - [field, value, method] if querying one field
+            - [ [field, value, method], [field, value, method] ] if multiple field query is desired
+        required: true
+        type: list
+    method:
+        description: set global method for conditions
+        required: false
+        type: string
+    plan_status:
+        description: pass "C" to query closed incidents
+        required: false
+        type: string
+    multiple_fields:
+        description: "true"/"True"/etc. to perform multifield query.
+        required: false
+        type: string
+    fail:
+        description: provide any value to fail the module
+        required: false
+        type: string
+    
+
+author:
+    - Brian Reid (@breid1313)
+'''
+
+#TODO
+EXAMPLES = r'''
+# Query open incidents by name, non-exact match
+- name: Test query on CP4S cases
+  ryan_gordon1.cloud_pak_for_security.cp4s_query_incidents:
+    conditions: ["name", "example_name", "contains"]
+
+# Query open incidents by name, exact match
+- name: Test query on CP4S cases
+  ryan_gordon1.cloud_pak_for_security.cp4s_query_incidents:
+    conditions: ["name", "example_name", "equals"]
+
+# Query closed incidents by name
+- name: Test query on CP4S cases
+  ryan_gordon1.cloud_pak_for_security.cp4s_query_incidents:
+    conditions: ["name", "example_name", "equals"]
```

Comment:
is this like a [key, value, operator] syntax?

Reply:
basically. to query cases in resilient on a condition, you need to specify an incident field ie name or description, a value, and a method ie equals for an exact match.

I updated some of the doc terminology, so this should be a bit more clear. I've also directed the user to the resilient/cases rest API docs, as they will need that and/or previously development experience with the product to know whats available to them